doc/csharp/csharp-null-ref-types-proposal.md

```diff
@@ -0,0 +1,197 @@
+ Handling of null reference types in C# code generation
+----
+* Author(s): Tony Newell
+* Approver: 
+* Status: Draft
+* Last updated: June 2023
```

Comment:
According to the guidelines in the proposal process, for each proposal we need to open a discussion thread in the grpc.io forum so that community has a chance to voice their opinions.
The process: https://github.com/grpc/proposal/blob/master/README.md#process (see bullet 5)
Example: https://github.com/grpc/proposal/blob/master/L103-core-move-insecure-creds-declaration.md